web_external/views/body/LayersPanel.js

```diff
@@ -32,16 +32,18 @@ const LayersPanel = Panel.extend({
     downloadGeojsonEvent: function (event) {
         var datasetId = $(event.currentTarget).attr('m-dataset-id');
         var dataset = this.collection.get(datasetId);
-        dataset.once('m:dataset_geo_dataLoaded', function (dataset) {
-            var data = dataset.get('geoData');
-            var a = window.document.createElement('a');
-            a.href = window.URL.createObjectURL(new Blob([JSON.stringify(data)], {type: 'application/json'}));
-            var filename = dataset.get('name') + '.geojson';
-            a.download = filename;
-            document.body.appendChild(a);
-            a.click();
-            document.body.removeChild(a);
-        }).loadGeoData();
+        dataset
+            .loadGeoData()
+            .then(() => {
+                var data = dataset.get('geoData');
+                var a = window.document.createElement('a');
+                a.href = window.URL.createObjectURL(new Blob([JSON.stringify(data)], { type: 'application/json' }));
+                var filename = dataset.get('name') + '.geojson';
```

Comment:
A minor improvement (which could be in a different PR): I have some datasets I uploaded called `<name>.geojson`, when I download the, I end up with a file called `<name>.geojson.geojson`.  If the dataset name ends with `.geojson`, we shouldn't append a second `.geojson` to its name.